pkgs/development/python-modules/fastapi/default.nix

```diff
@@ -10,24 +11,34 @@
 , pyjwt
 , passlib
 , aiosqlite
+, peewee
 }:
 
 buildPythonPackage rec {
   pname = "fastapi";
-  version = "0.45.0";
+  version = "0.49.0";
   format = "flit";
   disabled = !isPy3k;
 
   src = fetchFromGitHub {
     owner = "tiangolo";
     repo = "fastapi";
     rev = version;
-    sha256 = "1qwh382ny6qa3zi64micdq4j7dc64zv4rfd8g91j0digd4rhs6i1";
+    sha256 = "1dw5f2xvn0fqqsy29ypba8v3444cy7dvc7gkpmnhshky0rmfni3n";
   };
 
+  # Need Starlette version 0.12.9 for now due to
+  # https://github.com/tiangolo/fastapi/issues/683
+
   propagatedBuildInputs = [
     uvicorn
-    starlette
+    (starlette.overridePythonAttrs (old: rec {
+      version = "0.12.9";
+      src = fetchurl {
+        url = "https://github.com/encode/starlette/archive/${version}.tar.gz";
+        sha256 = "0d3g44hajqmjlr46mf527algqf6a5zh7k7dssl6hwzqm71qr09jp";
+      };
+    }))
```

Comment:
not okay. this will cause issues if another package brings in starlette at latest. You should patch instead you want to patch the version.

You should be able to be within a major version, and not be broken (on packages that adhere to semver)

example:
https://github.com/NixOS/nixpkgs/blob/24720dd250cb26025d6dd5dbaf089d48f2493cb5/pkgs/development/python-modules/celery/default.nix#L14

Reply:
Thanks for reviewing. Unfortunately, Fastapi is [completely broken](https://github.com/tiangolo/fastapi/issues/683#issuecomment-585931388) for starlette > 0.12.13. Thus, using the latest version of Starlette with Fastapi. I have three choices,

 - revert the version of Starlette in Starlette's own default.nix to be something compatible (Fastapi is Starlette's only dependent)
 - leave this broken for now until fastapi works correctly with later versions of Starlette
 - remove fastapi from Nixpkgs until fixed properly

I think I'll go with the first one unless there's an objection.

Reply:
reverting is fine, just unfortunate. I think this is reasonable.